azb2c-with-parcel.md

```diff
@@ -0,0 +1,106 @@
+Let's get our hands dirty and add the dependency to the current project (or create a new project first with ```yarn init```) ```yarn add --dev parcel``` 
+
+Next, I will create a new template file under ```/src/en.html``` (using en to represent the template for english) for azure b2c to be used:
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <title>My Product Brand Name</title>
+</head>
+<body>
+    <div id="api"></div>
+</body>
+</html>
+```
+
+In the ```package.json``` add a new line ```"start-b2c": "parcel ./src/*.html", ``` in the scripts section.
+Now start up your application via console ```yarn start-b2c```  and access the template on  [http://localhost:1234/en.html](http://localhost:1234/en.html)
+
+Congratulations, you have a basic template that would already be working.
+
+
+## HTML
+Now the HTML above can be extended with additional structures to fulfill the design as desired. A [sample version](https://github.com/csalv22/az-b2c-parcel/blob/main/src/en.html) with a bit more features can be found, but writing html & scss is not what this blog post is about.
+
+## CSS
+Now to the interesting part: Before writing the css styles for the templates, there are a few things we should do before.
+
+### SASS / SCSS
+We will rely on [sass](https://sass-lang.com/) as preprocessor to have a stronger toolset at hand than just plain css. This can be [added to our setup](https://parceljs.org/languages/sass/) very easily: ```yarn add --dev @parcel/transformer-sass```
+
+Create a new stylesheet in ```/src/scss/index.scss``` with the following content
+```scss
+div {
+  &#api {
+    width: 100px;
+    height: 100px;
+    background-color: #123456;
+  }
+}
+```
+
+Restart the developer process (```yarn start-b2c```) to apply the SCSS style you created.
+Now you're set up to design your html/css structure and can verify the result in the browser.
+
+### CSS normalize / reset
+As still all browsers render unstyled html elements slightly differently, the current web site would result in different appearance on the different browsers, which we have to address.
+I decided to go for [normalize.css](github.com/necolas/normalize.css) as I prefer having some basic styles in contrary to a reset css. There are many options and alternatives out there (e.g. [the-new-css-reset](https://www.npmjs.com/package/the-new-css-reset), [sanitize.css](https://csstools.github.io/sanitize.css/), [normalize.css](https://csstools.github.io/normalize.css/),...) to be explored by yourself.
+
+### CSS transpiling
+[Transpiling](https://parceljs.org/languages/css/#transpilation) is done out of the box by parcel.js, so if you have an appropriate configuration in your package.json, no additional steps are required.
+
+## JS
+In my use case I didn't have the need to have any custom JS, plain html & css serves our needs. But parcel.js gives you many options into hand (you could even go for one of the big JS Frameworks and use static site rendering to create your html templates)
```

Comment:
Would be next.js a solution? Maybe some hints where to have a look, some URLs etc.